framework/doc/content/source/meshgenerators/TransfiniteMeshGenerator.md

```diff
@@ -0,0 +1,70 @@
+
+The default distribution of points on an edge is equidistant and nothing expplicit needs to be specified. However, for meshes that need to capture more scales in different regions we have two additional options that can be prescribed.  The distribution of points on an edge can be either provided as a discrete set of points, using the edge type `DISCRETE`, or as a bias along the edge. The only requirement is that opposite edges have the same number of points.
+
+## Example 
+
+
+!media large_media/framework/meshgenerators/transfinite_arccircle_annular_sm.png style=float:right;margin:2%;width:18%; caption=Fig. 3: Annulus section generated using `CIRCARC` edges.
+
+
+&nbsp;
+
+This mesh generator considers 4 types of edges defined in `edge_type`, where `edge` is either left, right, top or bottom.
+
+  - `LINE`, constructs a line between provided corner vertices. A quadrilater can be obtained only with line sides as in Fig. 2.
+
+  - `CIRCARC`, constructs an arc of a circle and requires an additional parameter, named `edge_parameter`. This parameter can be the distance at the middle of an edge between a straight line and the arc of circle desired, as in the test [arccircle_generator.i](test/tests/meshgenerators/transfinite_generator/arcircle_generator.i). The edge is then generated by computing the middle point on the edge. The deviation from a straight line is set by convention to be positive if the arc circle is outward, or negative if inward, in this fashion the user can specify different orientations, as in Fig. 4.
+  The midpoint could also be obtained from a QUAD8 element as provided by an external mesh generator and inserted as input by the user in the same  `edge_parameter`, as in the test [arccircle_midpoint_generator.i](test/tests/meshgenerators/transfinite_generator/arcircle_midpoint_generator.i). Different point distributions along arc circles can be prescribed as for the other edge types and propagate via the polar coordinates parametrization.
+
+&nbsp;
+
+!media large_media/framework/meshgenerators/transfinite_discrete_arccircle_sm.png style=float:right;margin:2%;width:30%; caption=Fig. 4: A domain with edge types `CIRCARC` and `DISCRETE`.
+
+
+!listing test/tests/meshgenerators/transfinite_generator/discrete_generator.i
+    block=Mesh/transf
+
+
+  - `DISCRETE`, constructs a line along a set of points provided by the user in the order represented by arrows in Fig. 1, a test is available in [discrete_generator.i](test/tests/meshgenerators/transfinite_generator/discrete_generator.i) and illustrated in Fig. 4. This option is the only one that does not support a different point redistribution since it is assumed the user desires those edge points explicitely. Along an edge described by this option the `bias_x` and `bias_y` parameters will have no effect. 
```

Comment:
```suggestion
  - `DISCRETE`, constructs a line along a set of points provided by the user in the order represented by arrows in Fig. 1, a test is available in [discrete_generator.i](test/tests/meshgenerators/transfinite_generator/discrete_generator.i) and illustrated in Fig. 4. This option is the only one that does not support a different point redistribution since it is assumed the user desires those edge points explicitly. Along an edge described by this option the `bias_x` and `bias_y` parameters will have no effect. 
```